pkg/repository/prisma/dbsqlc/queue.sql

```diff
@@ -0,0 +1,80 @@
+-- name: UpsertQueue :exec
+INSERT INTO
+    "Queue" (
+        "tenantId",
+        "name"
+    )
+VALUES
+    (
+        @tenantId::uuid,
+        @name::text
+    )
+ON CONFLICT ("tenantId", "name") DO NOTHING;
+
+-- name: ListQueues :many
+SELECT
+    *
+FROM
+    "Queue"
+WHERE
+    "tenantId" = @tenantId::uuid;
+
+-- name: CreateQueueItem :exec
+INSERT INTO
+    "QueueItem" (
+        "stepRunId",
+        "stepId",
+        "actionId",
+        "scheduleTimeoutAt",
+        "stepTimeout",
+        "priority",
+        "isQueued",
+        "tenantId",
+        "queue",
+        "sticky",
+        "desiredWorkerId"
+    )
+VALUES
+    (
+        sqlc.narg('stepRunId')::uuid,
+        sqlc.narg('stepId')::uuid,
+        sqlc.narg('actionId')::text,
+        sqlc.narg('scheduleTimeoutAt')::timestamp,
+        sqlc.narg('stepTimeout')::text,
+        COALESCE(sqlc.narg('priority')::integer, 1),
+        true,
+        @tenantId::uuid,
+        @queue,
+        sqlc.narg('sticky')::"StickyStrategy",
+        sqlc.narg('desiredWorkerId')::uuid
+    );
+
+-- name: ListQueueItems :batchmany
+SELECT
+    *
+FROM
+    "QueueItem" qi
+WHERE
+    qi."isQueued" = true
+    AND qi."tenantId" = @tenantId::uuid
+    AND qi."queue" = @queue::text
+    AND (
+        sqlc.narg('gtId')::bigint IS NULL OR
+        qi."id" >= sqlc.narg('gtId')::bigint
+    )
+    -- TODO: verify that this forces index usage
+    AND qi."priority" >= 1 AND qi."priority" <= 4
+ORDER BY
+    qi."priority" DESC,
+    qi."id" ASC
+LIMIT
+    100
```

Comment:
should we pre-empt make this dynamic so we can further tune?

Reply:
good call, just added `SERVER_SINGLE_QUEUE_LIMIT`